x-pack/plugins/security/server/authentication/authentication_service.ts

```diff
@@ -216,6 +217,73 @@ export class AuthenticationService {
         },
       });
     });
+
+    elasticsearch.setUnauthorizedErrorHandler(async ({ error, request }, toolkit) => {
+      if (!this.authenticator) {
+        this.logger.error('Authentication sub-system is not fully initialized yet.');
+        return toolkit.notHandled();
+      }
+
+      if (!license.isLicenseAvailable() || !license.isEnabled()) {
+        this.logger.error(
+          `License is not available or does not support security features, re-authentication is not possible (available: ${license.isLicenseAvailable()}, enabled: ${license.isEnabled()}).`
+        );
+        return toolkit.notHandled();
+      }
+
+      if (getErrorStatusCode(error) !== 401) {
+        this.logger.error(
+          `Re-authentication is not possible for the following error: ${getDetailedErrorMessage(
+            error
+          )}.`
+        );
+        return toolkit.notHandled();
+      }
+
+      this.logger.warn(`Re-authenticating request due to error: ${getDetailedErrorMessage(error)}`);
```

Comment:
We're seeing this behavior pretty regularly for some users, and AFAIK there's nothing "wrong" with their configuration. So maybe this should be a `debug` log level instead of `warn`?

Reply:
Yeah, that's good point - it was useful during development, but can be quite confusing in production.